certbot/plugins/dns_common_lexicon.py

```diff
@@ -0,0 +1,94 @@
+"""Common code for DNS Authenticator Plugins built on Lexicon."""
+
+import logging
+
+from requests.exceptions import HTTPError, RequestException
+
+from certbot import errors
+from certbot.plugins import dns_common
+
+logger = logging.getLogger(__name__)
+
+
+class LexiconClient(object):
+    """
+    Encapsulates all communication with a DNS provider via Lexicon.
+    """
+
+    provider = None
```

Comment:
While this is technically fine, I think this should be an instance variable rather than a class variable based on how it's used.